### PR TITLE
Fix ImageAnnotation fill not updating after initial message

### PR DESCRIPTION
**Pull request by snosenzo**
**User-Facing Changes**

 - Fix ImageAnnotation fill not updating after initial message
**Description**

FG-3260

### CODE REVIEW

**Review by jtbandes — APPROVED** [review]
Thanks!

**Comment by jtbandes** [comment]
Should we try to test this in a storybook? Like send one message and then in the next tick send a 2nd annotations message?

**Comment by snosenzo** [comment]
Found another issue where if the geometry line buffer was previously larger, then the next annotation would use points from the previous geometry. Caught it when iterating with update story.

File: packages/studio-base/src/panels/ThreeDeeRender/renderables/ImageMode/annotations/RenderableLineAnnotation.ts [hunk 1]
```diff
@@ -180,7 +180,7 @@ export class RenderableLineAnnotation extends Renderable<BaseUserData, /*TRender
       if (
         this.#numPoints == undefined ||
         !this.#geometry ||
-        pointsLength > this.#numPoints ||
+        pointsLength !== this.#numPoints ||
```
Comment (jtbandes) [inline packages/studio-base/src/panels/ThreeDeeRender/renderables/ImageMode/annotations/RenderableLineAnnotation.ts:183]: was this the source of a bug? I think we should be able to reuse the geometry if it's larger than necessary, right?

Comment (jtbandes) [reply packages/studio-base/src/panels/ThreeDeeRender/renderables/ImageMode/annotations/RenderableLineAnnotation.ts:183]: Maybe need to use drawRange?

Comment (snosenzo) [reply packages/studio-base/src/panels/ThreeDeeRender/renderables/ImageMode/annotations/RenderableLineAnnotation.ts]: Figured it out. I tried using drawRange and that didn't work. Then I realized the instanceCount stuff and we were calculating that incorrectly. Should represent the number of lines and our counts for polygons and line strips were wrong.

Comment (jtbandes) [reply packages/studio-base/src/panels/ThreeDeeRender/renderables/ImageMode/annotations/RenderableLineAnnotation.ts]: Thanks for fixing!

**Pull request merged**